libraries/Microsoft.Bot.Builder.AI.LuisV3/ListElement.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Builder.AI.Luis
+{
+    /// <summary>
+    /// Defines a sub-list to append to an existing list entity.
+    /// </summary>
+    public class ListElement
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ListElement"/> class.
+        /// </summary>
+        public ListElement()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ListElement"/> class.
+        /// </summary>
+        /// <param name="canonicalForm">The canonical form of the sub-list.</param>
+        /// <param name="name">The name of the sub-list.</param>
+        /// <param name="synonyms">The synonyms of the canonical form.</param>
+        public ListElement(string canonicalForm, string name = null, IList<string> synonyms = null)
```

Comment:
You're making the synonyms here optional but you still validate them in the `Validate` method. As far as I remember, we don't allow explicitly sending synonyms as null, but allow not sending it at all. So if you add `NullValuehandling.Ignore` to the Synonyms property, null would be a valid value here.

Reply:
I've made synonyms as required.  The validation will catch if synonyms is null before sending--there is no point in a list with just a canonical form, nothing will match.

Reply:
BTW, what is the purpose of the "name" field on an individual element?  Doesn't really seem of any value and I'm wondering if I should drop it.

Reply:
Right now, it doesn't serve anything. I'm not sure why it's there. As for the synonyms, the canonical form itself is matched against. You don't always have synonyms for a phrase, e.g. when you pass in a list of emails.

Reply:
OK, I've removed the name and relaxed the validation on synonyms.